ldap-client.go

```diff
@@ -438,6 +456,27 @@ func (lc *LDAPClient) ChangeOpenLDAPUserPassword(username, oldPassword, newPassw
 	return nil
 }
 
+func (lc *LDAPClient) GetUserByCN(userCN,uidAttr string)(uid string, err error){
+	searchRequest := ldap.NewSearchRequest(
+		userCN,
+		ldap.ScopeWholeSubtree, ldap.NeverDerefAliases, 0, 0, false,
+		"(objectClass=*)",
+		[]string{"cn", uidAttr},
+		nil,
+	)
+
+	sr, err := lc.Conn.Search(searchRequest)
+	if err != nil {
+		return "", err
+	}
+	if len(sr.Entries) != 1 {
+		return "", nil
+	}
+	userEntry := sr.Entries[0]
+	uid = userEntry.GetAttributeValue(uidAttr)
```

Comment:
can it fail? (I assume not, but double checking)

Reply:
cannot fail